File: src/main/java/com/borvalcode/sudoku/infrastructure/gateway/sudokuboardapi/vo/GenerateResponse.java
package com.borvalcode.sudoku.infrastructure.gateway.sudokuboardapi.vo;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.Data;

@Data
public class GenerateResponse {

  private Response response;

  @Data
  public static class Response {

    private String difficulty;

    private List<List<Integer>> solution;

    @JsonProperty("unsolved-sudoku")
    private List<List<Integer>> unsolvedSudoku;
  }
}

File: src/main/kotlin/com/borvalcode/sudoku/infrastructure/gateway/sudokuboardapi/vo/GenerateResponse.kt
package com.borvalcode.sudoku.infrastructure.gateway.sudokuboardapi.vo

import com.fasterxml.jackson.annotation.JsonProperty

class GenerateResponse(val response: Response) {

    class Response(val difficulty: String, val solution: List<List<Int>>, @JsonProperty("unsolved-sudoku")
    val unsolvedSudoku: List<List<Int>>)
}